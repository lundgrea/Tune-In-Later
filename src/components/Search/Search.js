import React, {Component} from 'react';
import { connect } from 'react-redux';
import { addAlbums } from '../../actions'

class Search extends Component {
  constructor(){
    super();
    this.state = {
      search: ''
    }
  }

  handleChange = e => {
    this.setState({search: e.target.value})
  }

  fetchAlbums = e => {
    e.preventDefault();
    fetch(`https://itunes.apple.com/search?term=${this.state.search}&entity=album`)
    .then(res => res.json())
    // .then(data => console.log(data.results))
    .then(data => this.props.addAlbums(data.results))
    .catch(err => console.log(err))
    this.setState({ search: ''})
  }

  render() {
      return (
        <form>
          <input
            type="text"
            placeholder="Input Artist Name To Search"
            value={this.state.search}
            onChange={this.handleChange}
          />
          <button onClick={this.fetchAlbums}>SEARCH</button>
        </form>
      )
  }
}

const mapDispatchToProps = dispatch => ({
  addAlbums: albums => dispatch(addAlbums(albums))
})

export default connect(null, mapDispatchToProps)(Search)