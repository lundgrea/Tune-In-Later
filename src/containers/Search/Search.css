.search-form-inner {
  display: flex;
  flex-direction: column;
  width: 100%;
  align-items: center;
}

.search-button {
  width: 313.5px;
  height: 40px;
  font-size: 16px;
  padding: 10px;
  border: none;
  border-radius: 8px;
}

.search-input {
  width: 100%;
  font-size: 16px;
  padding: 10px;
  margin: 10px 10px 10px 10px;
  border: none;
  border-radius: 8px;
}
