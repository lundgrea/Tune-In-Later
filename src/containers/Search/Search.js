import React, {Component} from 'react';
import { connect } from 'react-redux';
import { addAlbums } from '../../actions';
import { getAlbums } from '../../apiCalls/apiCalls';
import { handleErrors } from '../../actions/index';
import { getFavorites } from '../../apiCalls/apiCalls';
import { Link } from "react-router-dom";
import PropTypes from 'prop-types'

export class Search extends Component {
  constructor(){
    super();
    this.state = {
      search: ''
    }
  }

  handleChange = e => {
    this.setState({search: e.target.value})
  }

  checkFavorites = async () => {
    const userFavorites = await getFavorites(this.props.user.id)
    const favoritesObject = userFavorites.includes(favorite => favorite.album_id === this.props.album_id)
    return favoritesObject
  }

  fetchAlbums = (e) => {
    e.preventDefault()
    getAlbums(this.state.search)
    .then(data => data.results.map(result => ({
      artist_name:  result.artistName,
      album_name:  result.collectionName,
      primary_genre_name:  result.primaryGenreName,
      album_id:  result.collectionId,
      artwork_url:  result.artworkUrl100,
      release_date: result.releaseDate,
      content_advisory_rating: result.contentAdvisoryRating,
      key: result.collectionId,
      isFavorite: false
    })))
    .then(data => this.props.addAlbums(data))
    .catch(err => console.log(err))
    this.setState({ search: ''})
    this.props.handleErrors('')
    
  }

  render() {
      return (
        <form>
          <input
            type="text"
            placeholder="Input Artist Name To Search"
            value={this.state.search}
            onChange={this.handleChange}
          />
          <Link to='/'>
            <button onClick={this.fetchAlbums}>SEARCH</button>
          </Link>
        </form>
      )
  }
}

const mapStateToProps = store => ({
  albums: store.albums,
  user: store.user
})

const mapDispatchToProps = dispatch => ({
  addAlbums: albums => dispatch(addAlbums(albums)),
  handleErrors: (error) => dispatch(handleErrors(error))
})

export default connect(mapStateToProps, mapDispatchToProps)(Search)

Search.propTypes = {
  albums: PropTypes.array.isRequired,
  user: PropTypes.object.isRequired,
  addAlbums: PropTypes.func.isRequired,
  handleErrors: PropTypes.func.isRequired
}