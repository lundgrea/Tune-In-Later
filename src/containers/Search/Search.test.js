import React from 'react';
import { shallow } from 'enzyme';
import { Search } from './Search';

describe('Search', () => {
  let wrapper;

  const mockUser = {
    id: 1, name: 'aidan', email: 'aidanmckay2000@gmail.com'
  }
  const mockAlbums = [
    {album_name: 'mirrorland', album_id:12345},
    {album_name: 'robots', album_id:123456},
  ]
  beforeEach(() => {
    wrapper = shallow(
      <Search
      albums={mockAlbums}
      user={mockUser}
      addAlbum={jest.fn()}
      handleError={jest.fn()}
      />
    )
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  it('should call handleChange method when change happens', () => {
    const mockEvent = {target: { value: 'earthgang'}}

    wrapper.find('input').simulate('change', mockEvent)
    expect(wrapper.state('search')).toEqual('earthgang')
  });

  it('should fetch albums', () => {
    const mockEvent = {}
    wrapper.find('button').simulate('click', mockEvent)
    
  });
});