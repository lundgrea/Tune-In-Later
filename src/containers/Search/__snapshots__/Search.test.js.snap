// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match snapshot 1`] = `
<div>
  <form
    className="search-form-inner"
  >
    <input
      className="search-input"
      onChange={[Function]}
      placeholder="Input Artist Name To Search"
      type="text"
      value=""
    />
    <Link
      to="/"
    >
      <button
        className="search-button"
        onClick={[Function]}
      >
        search
      </button>
    </Link>
  </form>
</div>
`;
